src/main/java/de/hpi/bpt/argos/api/entity/EntityEndpointImpl.java

```diff
@@ -0,0 +1,396 @@
+
+        sparkService.get(EntityEndpoint.getEventsOfEntityBaseUri(),
+				(Request request, Response response) ->
+						endpointUtil.executeRequest(logger, request, response, this::getEventsOfEntity));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getEntity(Request request, Response response) {
+        long entityId = getEntityId(request);
+		Entity entity;
+
+        if (entityId < 0) {
+			entity = VirtualRoot.getInstance();
+		} else {
+        	entity = PersistenceAdapterImpl.getInstance().getEntity(entityId);
+		}
+
+        if (entity == null) {
+        	halt(HttpStatusCodes.NOT_FOUND, "cannot find entity");
+		}
+
+		ObjectWrapper<List<Attribute>> attributes = new ObjectWrapper<>();
+        WatchImpl.measure("get attributes",
+				() -> attributes.set(PersistenceAdapterImpl.getInstance().getAttributes(entityId)));
+
+        ObjectWrapper<List<TypeAttribute>> typeAttributes = new ObjectWrapper<>();
+        WatchImpl.measure("get type attributes",
+				() -> typeAttributes.set(PersistenceAdapterImpl.getInstance().getTypeAttributes(entity.getTypeId())));
+
+        ObjectWrapper<List<Pair<TypeAttribute, Attribute>>> joinedAttributes = new ObjectWrapper<>();
+        WatchImpl.measure("join attributes", () -> joinedAttributes.set(joinAttributes(typeAttributes.get(), attributes.get())));
+
+        ObjectWrapper<JsonObject> jsonEntity = new ObjectWrapper<>();
+        WatchImpl.measure("get entity json", () -> jsonEntity.set(getEntityJson(entity, joinedAttributes.get())));
+
+        return serializer.toJson(jsonEntity.get());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getChildEntities(Request request, Response response) {
+        long entityId = getEntityId(request);
+        long entityTypeId = getEntityTypeId(request);
+        List<Entity> childEntities = PersistenceAdapterImpl.getInstance().getEntities(entityId, entityTypeId);
+        List<TypeAttribute> typeAttributesToInclude = getAttributesTypeIdsToInclude(request);
+
+		ObjectWrapper<JsonArray> jsonEntities = new ObjectWrapper<>();
+		WatchImpl.measure("getEntitiesJson", () -> jsonEntities.set(getEntitiesJson(childEntities, typeAttributesToInclude)));
+
+        return serializer.toJson(jsonEntities.get());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getEventTypesOfEntity(Request request, Response response) {
+        long entityId = getEntityId(request);
+        boolean includeChildEvents = getIncludeChildEvents(request);
+
+        EntityHierarchyNode entityNode = HierarchyBuilderImpl.getInstance().getEntityHierarchyRootNode().findChildEntity(entityId);
+
+        if (entityNode == null) {
+        	halt(HttpStatusCodes.NOT_FOUND, "cannot find entity in hierarchy");
+		}
+
+		List<Long> entityIds = new ArrayList<>();
+
+        if (includeChildEvents) {
+		 	entityIds.addAll(entityNode.getChildIds());
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'entityNode' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2259)